broker-core/src/test/java/io/zeebe/broker/logstreams/processor/SkipFailingEventsTest.java

```diff
@@ -0,0 +1,215 @@
+  private KeyGenerator keyGenerator;
+  private TypedStreamEnvironment env;
+  private BlackList blackList;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+
+    streams =
+        new TestStreams(
+            tempFolder.getRoot(), closeables, serviceContainerRule.get(), actorSchedulerRule.get());
+
+    stream = streams.createLogStream(STREAM_NAME);
+    env = new TypedStreamEnvironment(streams.getLogStream(STREAM_NAME), output);
+
+    final AtomicLong key = new AtomicLong();
+    keyGenerator = () -> key.getAndIncrement();
+  }
+
+  @Test
+  public void shouldSkipFailingEvent() {
+    // given
+    final DumpProcessor dumpProcessor = spy(new DumpProcessor());
+    final ErrorProneProcessor processor = new ErrorProneProcessor();
+
+    streamProcessorControl =
+        streams.initStreamProcessor(
+            STREAM_NAME,
+            STREAM_PROCESSOR_ID,
+            DefaultZeebeDbFactory.DEFAULT_DB_FACTORY,
+            (db) -> {
+              final ZeebeState zeebeState = new ZeebeState(db);
+              blackList = zeebeState.getBlackList();
+
+              return env.newStreamProcessor()
+                  .zeebeState(zeebeState)
+                  .onEvent(
+                      ValueType.WORKFLOW_INSTANCE,
+                      WorkflowInstanceIntent.ELEMENT_ACTIVATING,
+                      processor)
+                  .onEvent(
+                      ValueType.WORKFLOW_INSTANCE,
+                      WorkflowInstanceIntent.ELEMENT_ACTIVATED,
+                      dumpProcessor)
+                  .build();
+            });
+    streamProcessorControl.start();
+
+    streams
+        .newRecord(STREAM_NAME)
+        .event(workflowInstance(1))
+        .recordType(RecordType.EVENT)
+        .intent(WorkflowInstanceIntent.ELEMENT_ACTIVATING)
+        .key(keyGenerator.nextKey())
+        .write();
+    streams
+        .newRecord(STREAM_NAME)
+        .event(workflowInstance(1))
+        .recordType(RecordType.EVENT)
+        .intent(WorkflowInstanceIntent.ELEMENT_ACTIVATED)
+        .key(keyGenerator.nextKey())
+        .write();
+
+    // other instance
+    streams
+        .newRecord(STREAM_NAME)
+        .event(workflowInstance(2))
+        .recordType(RecordType.EVENT)
+        .intent(WorkflowInstanceIntent.ELEMENT_ACTIVATED)
+        .key(keyGenerator.nextKey())
+        .write();
+    // when
+    streamProcessorControl.unblock();
+
+    doRepeatedly(
+            () ->
+                streams
+                    .events(STREAM_NAME)
+                    .filter(
+                        e ->
+                            Records.isEvent(
+                                e,
+                                ValueType.WORKFLOW_INSTANCE,
+                                WorkflowInstanceIntent.ELEMENT_COMPLETED))
+                    .findFirst())
+        .until(o -> o.isPresent())
+        .get();
+
+    doRepeatedly(() -> {}).until(o -> processor.getProcessCount() > 0);
```

Comment:
Is it necessary to do `doRepeatedly`? Since the previous check for `ELEMENT_COMPLETED` should only happen if the last event was processed, then we must have processed the error prone processor's event, so we could just check if the process count == 1.

Reply:
Was more an wip